dsd/constraints.py

```diff
@@ -623,20 +617,18 @@ def satisfies_sequence_constraints(self, sequence: str) -> bool:
         """
         return all(constraint(sequence) for constraint in self.sequence_constraints)
 
-    def graph_hamming_distance(self, steps, previous_sequence):
+    def find_sequence_neighbors(self, steps: int, previous_sequence: str) -> List[str]:
         """
-        Makes an adjacency list for the previous sequence to allow a sequence a certain Hamming distance
-        away to replace the previous sequence. Creates a single new key:value pair each time. The previous
-        sequence is stored as a key with neighbors as its values, so neighbors of future sequences to be
-        replaced can be checked against the list of keys to make sure old sequences are not reused.
+        Makes an adjacency list of sequences some Hamming distance away from the previous sequence.
+        The sequence that replaces the previous sequence will be randomly chosen from this list.
         """
-        self.adjacency_list[previous_sequence] = []
-        for sequence in self._sequences:
-            # counts number of differences between sequences
+        adjacency_list= []
```

Comment:
Can we rename `adjacency_list` to `neighbors`? Technically, "adjacency list" refers to the whole graph, not to one list of neighbors (which is a bad name since it's actually several lists, but that's the convention)